src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double DEFAULT_MULTIPLICATION_NUMBER = 1.5;
+    private static final int START_ARRAY_SIZE = 0;
+    private T[] elementsData;
+    private int capacity;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.elementsData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.capacity = DEFAULT_CAPACITY;
+        this.size = START_ARRAY_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == capacity) {
+            double newCapacity = capacity * DEFAULT_MULTIPLICATION_NUMBER;
+            this.capacity = (int) newCapacity;
+            T[] newElementsData = (T[]) new Object[capacity];
+            System.arraycopy(elementsData, 0, newElementsData, 0, size);
+            elementsData = newElementsData;
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size() == capacity) {
+            double newCapacity = capacity * DEFAULT_MULTIPLICATION_NUMBER;
+            this.capacity = (int) newCapacity;
+            T[] newElementsData = (T[]) new Object[capacity];
+            System.arraycopy(elementsData, 0, newElementsData, 0, size);
+            elementsData = newElementsData;
+        }
+        if (index >= 0 && index <= size) {
+            for (int i = elementsData.length; i > 0; i--) {
+                elementsData[i] = elementsData[i - 1];
+                if (i == index) {
+                    elementsData[i] = value;
+                    size++;
+                    break;
+                }
+            }
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
```

Comment:
The logic for adding an element at a specific index is incorrect. The throw statement is outside of the else block, which means it will always be executed after the if block, regardless of whether the index is valid or not. Also, use System.arraycopy to shift elements, and ensure you check the index before attempting to add the element.